### ISSUE TITLE
JsonFormatter attach text to last step when sceneario.write is invoked from after hook

### ISSUE THREAD

**Issue by fmarmar** [cucumber/cucumber-jvm#1080]
## Summary

If an after hook invoke the method scenario.write(), the text is written as an entry in the node 'output' of the last step instead of the hook step in 'after' node

## Expected Behavior

Given the following code:

```
@After
public void endScenario(Scenario scenario) {
   scenario.write("test")
}
```

The cucumber json report should be:

```
[
  {
    "line": 1,
    "elements": [
      {
        "before": [
          {
            "result": {
              "duration": 102032546,
              "status": "passed"
            },
            "match": {
              "location": "Hooks.startScenario(Scenario)"
            }
          }
        ],
        "line": 20,
        "name": "Example",
        "description": "",
        "id": "example;example",
        "after": [
          {
            "output": [
              "test"
            ],
            "result": {
              "duration": 1519175111,
              "status": "passed"
            },
            "match": {
              "location": "Hooks.endScenario(Scenario)"
            }
          }
        ],
        "type": "scenario",
        "keyword": "Scenario",
        "steps": [
          {
            "result": {
              "duration": 1282972237,
              "status": "passed"
            },
            "line": 39,
            "name": "Example test",
            "match": {
              "location": "Step.example"
            },
            "keyword": "Given"
          }
        ]
      }
    ]
  }
]
```

## Current Behavior

```
[
  {
    "line": 1,
    "elements": [
      {
        "before": [
          {
            "result": {
              "duration": 102032546,
              "status": "passed"
            },
            "match": {
              "location": "Hooks.startScenario(Scenario)"
            }
          }
        ],
        "line": 20,
        "name": "Example",
        "description": "",
        "id": "example;example",
        "after": [
          {
            "result": {
              "duration": 1519175111,
              "status": "passed"
            },
            "match": {
              "location": "Hooks.endScenario(Scenario)"
            }
          }
        ],
        "type": "scenario",
        "keyword": "Scenario",
        "steps": [
          {
            "output": [
              "test"
            ],
            "result": {
              "duration": 1282972237,
              "status": "passed"
            },
            "line": 39,
            "name": "Example test",
            "match": {
              "location": "Step.example"
            },
            "keyword": "Given"
          }
        ]
      }
    ]
  }
]
```

## Context & Motivation

Using an external tool to create an html report, I expect the text to be associated to the hook, not the las step

## Your Environment

* Version used: 1.2.4

**Comment by brasmusson** [comment]
Since this bug is in Gherkin v2.12.2, this issue will be fixed when Cucumber-JVM is updated to use Gherkin v4 (#1035).

**Comment by brasmusson** [comment]
Now when #1035 has been merged, the 2.0.0-SNAPSHOT version from the master branch (with the groupId io.cucumber), the Json Formatter handles writes from the after hooks correctly.
